### PR TITLE
Add a variable rate shading demo

### PR DESCRIPTION
**Pull request by Calinou**
- Follow-up to https://github.com/godotengine/godot-docs/pull/6402.

This project showcases how to use variable rate shading (VRS) in 3D to improve performance.

Performance metrics are also displayed to evaluate potential performance gains.

Once this PR is merged, we can link to the demo in the documentation :slightly_smiling_face: 

## Preview

variable_rate_shading webp

### CODE REVIEW

**Comment by fire** [comment]
Will be testing this today.

File: 3d/variable_rate_shading/vrs.gd [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			# Disable a lot of VR-specific stuff like lens distortion.
+			xr_interface.eye_height = 0.0
+			xr_interface.k1 = 0.0
+			xr_interface.k2 = 0.0
+			xr_interface.oversample = 1.0
+
+			get_viewport().use_xr = true
+			xr_camera.current = true
+
+			# Reposition our origin point to work around an engine bug.
```
Comment (fire) [inline 3d/variable_rate_shading/vrs.gd:26]: Is this bug still active?

Comment (Calinou) [reply 3d/variable_rate_shading/vrs.gd:26]: I commented out the line and the bug occurred, so yes.

File: 3d/variable_rate_shading/vrs.gd [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+func _update_texture() -> void:
+	var vrs_mode = get_viewport().vrs_mode
+	if vrs_mode == Viewport.VRS_DISABLED:
+		texture_rect.visible = false
+	elif vrs_mode == Viewport.VRS_TEXTURE:
+		get_viewport().vrs_texture = texture
+		texture_rect.texture = texture
+		texture_rect.visible = true
+	elif vrs_mode == Viewport.VRS_XR:
+		# Doesn't seem to be supported yet. This should be exposed in a future engine version.
```
[the remark this answers is not in the document]
Comment (Calinou) [reply 3d/variable_rate_shading/vrs.gd:45]: To my knowledge, yes.

cc @BastiaanOlij

File: 3d/variable_rate_shading/vrs.gd (hunk 1 above)
Comment (BastiaanOlij) [reply 3d/variable_rate_shading/vrs.gd:26]: Hmm, weird, I was under the impression that was fixed. I might need to have another look at that :)

File: 3d/variable_rate_shading/vrs.gd (hunk 2 above)
[the remark this answers is not in the document]
Comment (BastiaanOlij) [reply 3d/variable_rate_shading/vrs.gd:45]: I think this is just a matter of binding `get_vrs_texture` in `XRInterface` so we have access to it.

**Review by BastiaanOlij — APPROVED** [review]
I think there is room for improvement but for a first version of a demo that will give people some instruction on how to use this, it's good enough.

**Pull request merged**